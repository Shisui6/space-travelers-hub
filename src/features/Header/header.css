.header {
  border-bottom: 1px solid #bebfc5;
  top: 0;
  margin-bottom: 12px;
}

.logo {
  display: flex;
  align-items: center;
  justify-content: center;
}

.logo img {
  height: 56px;
  margin: 0 8px;
}

.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px 64px;
}

.nav-menu {
  display: flex;
}

.nav-link {
  margin: 0 8px;
}

.nav-link a {
  color: #025fac;
  text-decoration: none;
}

.nav-link-active {
  border-bottom: 1px solid #025fac;
}

.border-left {
  border-left: 2px solid #bebfc5;
  padding-left: 16px;
}

.list {
  list-style-type: none;
}
