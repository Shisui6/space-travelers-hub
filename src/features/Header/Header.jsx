import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../../space-logo.png';
import './header.css';

const Header = () => {
  const activeClassName = 'nav-link-active';
  return (
    <header className="header">
      <nav className="navbar">
        <div className="logo">
          <img src={logo} alt="logo" />
          <h2>Space Travelers&apos; Hub </h2>
        </div>
        <ul className="nav-menu list">
          <li className="nav-link">
            <NavLink
              to="/"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              Rockets
            </NavLink>
          </li>
          <li className="nav-link">
            <NavLink
              to="/missions"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              Missions
            </NavLink>
          </li>
          <li className="nav-link border-left">
            <NavLink
              to="/profile"
              className={({ isActive }) => (isActive ? activeClassName : undefined)}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
