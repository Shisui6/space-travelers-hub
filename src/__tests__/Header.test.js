import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../features/Header/Header';

describe('Header', () => {
  it('renders Header component', async () => {
    const tree = render(<Router><Header /></Router>);
    expect(tree).toMatchSnapshot();
  });

  it('contains the mission link', () => {
    render(
      <Router>
        <Header />
      </Router>,
    );

    expect(screen.getByText('Missions')).toBeInTheDocument();
  });
});
