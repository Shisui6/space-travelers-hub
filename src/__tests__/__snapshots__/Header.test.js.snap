// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <nav
          class="navbar"
        >
          <div
            class="logo"
          >
            <img
              alt="logo"
              src="space-logo.png"
            />
            <h2>
              Space Travelers' Hub 
            </h2>
          </div>
          <ul
            class="nav-menu list"
          >
            <li
              class="nav-link"
            >
              <a
                aria-current="page"
                class="nav-link-active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="nav-link"
            >
              <a
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="nav-link border-left"
            >
              <a
                href="/profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <nav
        class="navbar"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="space-logo.png"
          />
          <h2>
            Space Travelers' Hub 
          </h2>
        </div>
        <ul
          class="nav-menu list"
        >
          <li
            class="nav-link"
          >
            <a
              aria-current="page"
              class="nav-link-active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li
            class="nav-link"
          >
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="nav-link border-left"
          >
            <a
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
